src/UI/HoverView.re

```diff
@@ -3,97 +3,154 @@
+    backgroundColor(theme.editorHoverWidgetBackground),
+  ];
+};
+
+let%component hoverItem =
+              (
+                ~x,
+                ~y,
+                ~diagnostics,
+                ~buffer,
+                ~location,
+                ~delay,
+                ~theme,
+                ~editorFont,
+                (),
+              ) => {
+  let%hook (opacity, _, _) =
+    Animation.animate(Time.ms(250))
+    |> Animation.delay(delay)
+    |> Animation.tween(0., 1.)
+    |> Hooks.animation;
+
+  let diagnostics =
+    Diagnostics.getDiagnosticsAtPosition(diagnostics, buffer, location);
+
+  let width = {
+    let measure = text =>
+      int_of_float(EditorFont.measure(~text, editorFont) +. 0.5);
+    let maxElementWidth =
       List.fold_left(
-        (prev, curr: Model.Diagnostic.t) => {
-          let (prevWidth, prevHeight, prevDiags) = prev;
-
-          let message = curr.message;
-          let width =
-            EditorFont.measure(~text=message, editorFont)
-            +. 0.5
-            |> int_of_float;
-          let height =
-            EditorFont.getHeight(editorFont) +. 0.5 |> int_of_float;
-
-          let newWidth = max(prevWidth, width + padding);
-          let newHeight = height + prevHeight + innerPadding;
-          let newElem = <Text style=textStyle text=message />;
-          let newDiags = [newElem, ...prevDiags];
-          (newWidth, newHeight, newDiags);
-        },
-        (0, 0, []),
-        hoverInfo.diagnostics,
+        (maxWidth, {message, _}: Diagnostic.t) =>
+          max(maxWidth, measure(message) + Constants.padding),
+        0,
+        diagnostics,
       );
+    maxElementWidth + Constants.padding * 2;
+  };
+
+  let height = {
+    let fontHeight = int_of_float(EditorFont.getHeight(editorFont) +. 0.5);
+    let elementHeight = fontHeight + Constants.innerPadding;
+    elementHeight * List.length(diagnostics) + Constants.padding * 2;
+  };
+
+  let elements =
+    diagnostics
+    |> List.map(({message, _}: Diagnostic.t) =>
+         <Text style={Styles.text(~theme, ~editorFont)} text=message />
+       )
+    |> List.rev
+    |> React.listToElement;
+
+  <View style={Styles.outerPosition(~x, ~y)}>
+    <Opacity opacity>
+      <View style={Styles.innerPosition(~width, ~height, ~theme)}>
+        elements
+      </View>
+    </Opacity>
+  </View>;
+};
+
+let make = (~x, ~y, ~state: State.t, ()) => {
+  let delay =
+    Configuration.getValue(c => c.editorHoverDelay, state.configuration)
+    |> Time.ms;
+  let hoverEnabled =
+    Configuration.getValue(c => c.editorHoverEnabled, state.configuration);
+
+  let State.{theme, editorFont, diagnostics, _} = state;
+  let maybeEditor =
+    state |> Selectors.getActiveEditorGroup |> Selectors.getActiveEditor;
+
+  switch (maybeEditor) {
+  | Some(editor) when hoverEnabled && state.mode != Vim.Types.Insert =>
+    switch (Buffers.getBuffer(editor.bufferId, state.buffers)) {
+    | Some(buffer) =>
```

Comment:
Related to the issue / screenshot I logged - seems like we're missing the case where there is no 'hover' - we're always showing at least a blank container w/ this change.

Reply:
Oh man, how did I miss that! Fixed in https://github.com/onivim/oni2/pull/1201/commits/d8b825b7f1290e0e7bb923694bb7c9d606dc937e